lib/Service/ExternalFolderService.php

```diff
@@ -906,4 +905,38 @@ public function save(ExternalFolder $storage) {
 	public function remove(int $storageId) {
 		$this->externalFolderRequest->remove($storageId);
 	}
+
+
+	/**
+	 *
+	 */
+	public function purgeRestoringPoints() {
+		foreach ($this->getAll() as $external) {
+			try {
+				$this->purgeExternalRestoringPoints($external);
+			} catch (ExternalFolderNotFoundException $e) {
+				$this->e($e, ['external' => $external]);
+			}
+		}
+	}
+
+	/**
+	 * @throws ExternalFolderNotFoundException
+	 */
+	public function purgeExternalRestoringPoints(ExternalFolder $external) {
+		$c = $this->configService->getAppValue(ConfigService::STORE_ITEMS_EXTERNAL);
+		$i = 0;
+		foreach ($this->getRestoringPoints($external) as $point) {
+			if ($point->isArchive()) {
+				continue;
+			}
+			$i++;
+			if ($i > $c) {
```

Comment:
This logic leads to orphaned differential restoring points. (same as current internal logic)
Parents need to be kept until no depended restoring points are available.

Reply:
Please see my proposed replacement of this patch above. I eliminated the problem. I was hoping @ArtificialOwl would incorporate... didn't want to create a competing pull request.

Reply:
I have been thinking about this logic problem for a while now. 
It should be handled in an easy way so that everybody can understand it.
Also it should be the same logic for internal and external storage as long as we have only the setting for number of items to be kept.
I'm already working on such a solution for internal storage first.
If it works as intended I would also suggest to incorporate the adaption into @ArtificialOwl pull request.
It would be a good idea to discuss our solutions to find a consensus.

Reply:
I have been using the code I provided in the two different pull requests for over two months and everything has worked perfectly aside from the isArchive() issue you mentioned. Please look at https://github.com/jxmx/backup/tree/fix/147/purge-external-storage and Pull #183. I'm fairly confident the logic is solid and has been working in my environment.

Reply:
I have seen your PR, I will review it later today.